File: Leetcode_solutions/n-ary-tree-postorder-traversal.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
  if (root === null) {
    return [];
  }
  output = [];
  search(root.children, output);
  output.push(root.val);
  return output;
};

function search(r, o) {
  for (let i = 0; i < r.length; i++) {
    if (r[i].children.length === 0) {
      o.push(r[i].val);
    } else {
      search(r[i].children, o);
      o.push(r[i].val);
    }
  }
  return o;
}

File: Leetcode_solutions/N-ary Tree Postorder Traversal.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""


class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        output = []
        if root == None:
            return output
        if root.children == None:
            return [root.val]
        self.search(root.children, output)
        output.append(root.val)
        return output

    def search(self, node, ans):
        for i in range(len(node)):
            if node[i].children == None:
                ans.append(node[i].val)
            else:
                self.search(node[i].children, ans)
                ans.append(node[i].val)
        return ans
